middleware/auth.go

```diff
@@ -0,0 +1,32 @@
+package middleware
+
+import (
+	"net/http"
+	"strings"
+)
+
+func isAuthorized(r *http.Request) bool {
+	// TODO: Add configuration to enable this Auth
+	// This configuration should not be as command line option
+	// Because, one glusterd can be run with `--no-auth` and
+	// other glusterd with auth enabled.
+	if strings.HasPrefix(r.RemoteAddr, "127.0.0.1:") {
```

Comment:
We should stop matching network addresses as strings. We do that in GD1 and in protocol/auth, and it isn't really good.